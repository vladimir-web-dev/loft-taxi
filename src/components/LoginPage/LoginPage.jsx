import React from 'react';
import PropTypes from 'prop-types';
import LoginForm from '../LoginForm';


import logo from '../../images/logo.png';
import './LoginPage.css';


function LoginPage (props) {
    return (
        <section className='section login'>
            <div className='login__content'>
                <div className='login__left'> 
                    <img className='logo' src={logo} alt="Logo"></img>
                </div>
                <div className='login__right'>
                    <LoginForm changePage={props.changePage}/>
                </div>        
            </div>            
        </section>
    );
}

LoginPage.propTypes = {
    changePage: PropTypes.func.isRequired
}

export default LoginPage;