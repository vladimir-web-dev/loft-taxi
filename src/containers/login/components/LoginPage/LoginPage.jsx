import React from "react";
import { useSelector } from "react-redux";
import { LoginForm } from "../LoginForm";
import { getIsAuthenticatingSelector } from "../../store";
import { Logo } from "loft-taxi-mui-theme";
import containerStyles from  "../../../../css_modules/Container.module.css";


export function LoginPage({ history }) {
  const isAuthenticating = useSelector(state =>
    getIsAuthenticatingSelector(state.authReducer)
  );

  return (
    <section className={containerStyles.central}>
      <div className={containerStyles.content}>
        <div className={containerStyles.col}>
          <Logo className={containerStyles.logo}/>
        </div>
        <div className={containerStyles.col}>
          <LoginForm history={history} />
        </div>
      </div>
    </section>
  );
}