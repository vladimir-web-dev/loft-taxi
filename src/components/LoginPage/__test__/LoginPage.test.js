import React from 'react';
import ReactDOM from 'react-dom';
import LoginPage from '../LoginPage';
import {render} from '@testing-library/react';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LoginPage changePage={() => {}}></LoginPage>, div);
    ReactDOM.unmountComponentAtNode(div);
})

it('matches snapshot', () => {
    const {container} = render(<LoginPage changePage={() => {}}></LoginPage>)
    expect(container).toMatchSnapshot(); 
})