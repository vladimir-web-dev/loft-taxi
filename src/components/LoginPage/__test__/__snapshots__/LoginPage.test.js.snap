// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <section
    class="section login"
  >
    <div
      class="login__content"
    >
      <div
        class="login__left"
      >
        <img
          alt="Logo"
          class="logo"
          src="logo.png"
        />
      </div>
      <div
        class="login__right"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
          elevatio="1"
          style="padding: 48px 48px;"
        >
          <form
            class="formTag"
            data-testid="login-form"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <h1
                  class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
                >
                  Войти
                </h1>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                Новый пользователь? 
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
                  data-testid="link-register"
                >
                   Зарегистрируйтесь
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                  data-testid="input-name"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Имя пользователя
                    <span
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                      placeholder="Имя пользователя"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                  data-testid="input-pass"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Пароль
                    <span
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                      placeholder="Пароль"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-12"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                  data-testid="button-login"
                  tabindex="0"
                  type="submit"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Войти
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </section>
</div>
`;
